proposals/0351-regex-builder.md

```diff
@@ -302,6 +299,22 @@ Regex {
 } // Regex<(Substring, Int, Float, Substring)>
 ```
 
+The following example creates a regex by concatenating subpatterns.
+
+```swift
+let regex = Regex {
+  "regex"
+  "builder"
+  "is"
+  "easy"
+}
+let match = try regex.prefixMatch(in: "regex builder is easy!")
+match?.0 // => "regex builder is easy"
+```
+
+<details>
+<summary>API definition</summary>
```

Comment:
I think, in general at least, we want to hoist the struct decls out of the details triangle. The stuff to go in the triangle would be the speculation about how variadic generics work (if it even belongs in this proposal) and the result builder overloads and machinery.

Reply:
All struct declarations already have an abbreviated version outside the collapsible. Inside the collapsible there's a full declaration along with initialize roverloads.

Reply:
But why not just put the declaration outside? Stick the comments about what we would have done if variadic generics existed and the machinery inside the disclosure. I don't think we should be hiding end-developer API inside the triangles